actions/schema-up/index.js

```diff
@@ -0,0 +1,111 @@
+    {reject: false},
+  );
+
+  if (hasRelayChanges === 0 && !relayFailed) {
+    tools.log.info('Generated relay files are unchanged.');
+    await tools.runInWorkspace('git', ['push']);
+    tools.exit.success('Schema is up to date on master.');
+  }
+
+  tools.log.info('Checking for unmerged schema update pull requests.');
+  const {data: openSchemaUpdatePRs} = await tools.github.graphql(`
+    query OpenPullRequests($owner: String!, $repo: String!) {
+      repository(owner: $owner, name: repo) {
+        pullRequests(first: 1, states: [OPEN], labels: ["schema update"]) {
+          totalCount
+        }
+      }
+    }
+  `, {...tools.context.repo});
+
+  if (openSchemaUpdatePRs.repository.pullRequests.totalCount > 0) {
+    tools.exit.failure('One or more schema update pull requests are already open. Please resolve those first.');
+  }
+
+  const branchName = `schema-update/${Date.now()}`;
+  tools.log.info(`Commiting relay-compiler changes to a new branch ${branchName}.`);
+  await tools.runInWorkspace('git', ['checkout', '-b', branchName]);
+  if (!relayFailed) {
+    await tools.runInWorkspace('git', ['commit', '--all', '--message=":gear: relay-compiler changes"']);
+  }
+  await tools.runInWorkspace('git', ['push', 'origin', branchName]);
+
+  tools.log.info('Creating a pull request.');
+
+  let body = `:robot: _This automated pull request brought to you by [a GitHub action](/actions/schema-up)_ :robot:
+
+The GraphQL schema has been automatically updated and \`relay-compiler\` has been re-run on the package source.`;
+
+  if (!relayFailed) {
+    body += ' The modified files have been committed to this branch and pushed. ';
+    body += 'If all of the tests pass in CI, merge with confidence :zap:';
+  } else {
+    body += ' `relay-compiler` failed with the following output:\n\n```\n';
+    body += relayOutput;
+    body += '\n```\n\nCheck out this branch to fix things so we don\'t break.';
+  }
+
+  const {data: createPR} = await tools.github.graphql(`
+    mutation CreatePullRequest($owner: String!, $repo: String!, $body: String!, $headRefName: String!) {
+      createPullRequest(input: {
+        title: "GraphQL schema update"
+        body: $body
+        headRefName: $headRefName
+      }) {
+        pullRequest {
+          id
+          number
+        }
+      }
+    }
+  `, {...tools.context.repo, body, headRefName: branchName});
+
+  const createdPullRequest = createPR.createPullRequest.pullRequest;
+  tools.log.info(
+    `Pull request #${createdPullRequest.number} has been opened with the changes from this schema upgrade.`);
+
+  await tools.github.graphql(`
+    mutation LabelPullRequest($id: ID!, $labelIDs: [ID!]!) {
+      addLabelsToLabelable(input: {
+        labelableId: $id,
+        labelIds: []
+      }) {
+        labelable {
+          id
+        }
+      }
+    }
+  `, {id: createdPullRequest.id, labelIDs: ['MDU6TGFiZWwxMzQyMzM1MjQ2']});
+}, {
```

Comment:
Is it worth checking/logging that the label has been applied successfully? Maybe not a big deal...